api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java

```diff
@@ -283,6 +285,23 @@ public boolean isProviderIdentifierUnique(Provider provider) throws APIException
 	public Provider getProviderByIdentifier(String identifier) {
 		return dao.getProviderByIdentifier(identifier);
 	}
+
+	@Transactional(readOnly = true)
+	public Provider createProviderFromUser(User user) {
+		
+		   Provider p = new Provider();
+		
+               if (user == null || user.getPerson() != null) {
+			   throw new APIException("User cannot be null");
```

Comment:
```suggestion
			   throw new APIException("User was not found");
```
Better to give user-friendly error messages